### PR TITLE
Add tests for multi_host_url_schema() and definitions_schema()

### PR DESCRIPTION
**Pull request by vickumar1981**
## Change Summary

- Adds unit tests for `json_schema.py` functions `multi_host_url_schema()` and `definitions_schema()`

## Related issue number

No related issue

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

File: tests/test_json_schema.py [hunk 1]
```diff
@@ -3459,3 +3459,46 @@ class OuterModel(BaseModel):
         'title': 'OuterModel',
         'type': 'object',
     }
+
+
+def test_multi_host_url_schema():
+    class TestGenerateJsonSchema(GenerateJsonSchema):
```
Comment (hramezani) [inline tests/test_json_schema.py:3465]: Why do we need this class here?

**Comment by hramezani** [comment]
Thanks @vickumar1981 for the path.

I've added some comments that are valid for other tests as well(I didn't add same comments for other tests). let's wait for @samuelcolvin to see what he thinks

File: tests/test_json_schema.py (hunk 1 above)
Comment (vickumar1981) [reply tests/test_json_schema.py]: @hramezani   Removed the unneeded class in the 3 tests.

File: tests/test_json_schema.py [hunk 2]
```diff
@@ -3459,3 +3459,46 @@ class OuterModel(BaseModel):
...
+        pass
+
+    schema = core_schema.multi_host_url_schema()
```
[the remark this answers is not in the document]
Comment (samuelcolvin) [reply tests/test_json_schema.py:3465]: yes, as @hramezani says, we should create a model, then construct a schema for this, this approach won't test the missing code.

[the remark this answers is not in the document]
Comment (samuelcolvin) [reply tests/test_json_schema.py:3465]: same for the examples below.

**Comment by samuelcolvin** [comment]
also, looks like there are conflicts, please update.

**Comment by samuelcolvin** [comment]
closing as no response.

**Pull request closed without merging**